### ISSUE TITLE
Plans to sell pre-made vents?

### ISSUE THREAD

**Issue by Daniel-dev22** [BrobstonCreations/yet-another-smart-vent#21]
Any plans to sell pre-assembled vents? I would be interested in buying a bunch.

**Comment by TonyBrobston** [comment]
Hey @daniel-dev22 , thanks for reaching out.

I've been working towards offering assembled units and am getting pretty close to being ready to.

The last piece I would like to test and verify before offering an assembled unit is related to the furnace and whether the plastics will deform over time from the heat. The filament I have been printing with is [3DFuel Pro PLA+](https://www.3dfuel.com/products/pro-pla-midnight-black-1-75mm) which has a heat resistance of [85C or 185F](https://cdn.shopify.com/s/files/1/0027/5339/6848/files/3D870-charts-min_600x600.jpg?v=1614282114).

I haven't had an opportunity to adequately verify because the weather hasn't been cold enough to turn my heat on.

From a few internet searches, it sounds like the typical temperature rise for a furnace is 40-70F. Assuming that is a valid temperature rise for a furnace, the return air would need to be 115F to produce a supply air of 185F. That temperature would be measured 1-2 meters after the furnace on the supply side and that temperature will drop as the air travels closer to the vent. 

In short, the vent is likely ready. But, I want to do my due diligence and hopefully that makes for happy customers.

Currently I'm thinking I need at least a few weeks of colder weather to run my furnace and then I should feel more comfortable selling this solution. At that point I plan to add a listing to [my store](https://brobston-creations.mybigcommerce.com/) and will probably use a preorder model as I ramp up production.

I appreciate your interest and am excited to get this offering out there as soon as possible.

**Comment by TonyBrobston** [comment]
> Awesome I appreciate the due-dilligence, do you plan to offer them in all the sizes you have 3d models for on the readme?

Yes. The hope would be to offer all the sizes I currently have models for. I imagine over time there will be other sizes added too, as people request less common ones.

I would prefer to only sell vents with one-piece shells, I created some snap-together solutions because of print bed size constraints, so I may start with 10-inch wide vents, while I acquire a larger printer and get it up and running.

The other thing that's hard to predict is how much demand there will or won't be. 3d printers are great for decentralized manufacturing, prototyping, and lots of one-offs. But another solution would likely be needed for high throughput if demand is high; that may have an effect on what sizes I offer, but we'll cross the bridge only if we end up there.

> Also, how would we find out when you're ready to start selling pre-orders on your store?

I will do my best to make a post here when I add a listing to my store. Another option would be to "watch" this repo and I think GitHub will notify you. I plan to update the readme with a link to the listing on my store.

It looks like the store I'm using also has a newsletter setup ([at the bottom of the page](https://brobston-creations.mybigcommerce.com/)), I may use that someday.

> Lastly, any chance for White vents or only black is possible?

Definitely. I had printed one prototype in white, I'll dig around and see if I have a picture.

I will likely offer other colors by request. These are all of the color options: https://www.3dfuel.com/collections/175ppla It's really just a matter of whether it makes financial sense to keep these stocked or order filament as orders come in and manage expectations that it may be a few weeks before a custom receives their product.

I'd also like to eventually offer wood tops. I did prototype out of Oak and Purple Heart. I'll add some pictures shortly. The CNC I'm currently using is not great for throughput, so I'd need to invest in another machine.

**Comment by TonyBrobston** [comment]
I accidentally hit the "Comment and close", but meant to keep this open.

**Comment by TonyBrobston** [comment]
As of yesterday, I ordered a larger 3D printer kit. Hopefully in the next few weeks it will be up and running.

This will enable me to print one piece vent tops and vent shells.

**Comment by veaceslavc** [comment]
Also interested in pre-made 4x10 vents. Subscribed to get a notification when available.
Thanks for the great work and diligence in implementing it.

**Comment by TonyBrobston** [comment]
Another update. I have gotten my larger printer kit together and have been able to print.

After some more review I realized I was looking at the annealed temperatures for 3DFuel's Pro Pla+, though annealing really isn't a great option because the printed part does not keep it's form after being annealed. I haven't had any issues with the vents I've printed, they've seen a month or two of heat so far. 

However I am going to move on to ABS, which will hopefully be the last change before offering assembled vents. I will soon be testing ABS on my larger printer. I expect some warping issues that I will hopefully be able to resolve with halogen lamps and/or a heated enclosure.

I wish things could move faster. I'm not a fan of making excuses, so they'll be ready as soon as they're ready. Just know that I'm working at it and you guys will be the first to know when I get there.

**Comment by TonyBrobston** [comment]
@Daniel-dev22 The dimensions above look right. My house also has mainly 4x10 vents and I believe the opening measures to roughly 4x10. In some cases I think they may be undersized or oversized, which is seems to be due to the opening cut in the sub-floor.

We are planning to list the vents on our website in the near future. I'll be sure to update you here when we do.

**Comment by TonyBrobston** [comment]
@Daniel-dev22 @veaceslavc @AwokenByMyFears 

My apologies for the radio-silence. I had taken a break from this project for a few months. In the last few weeks I've started working on it again.

We're excited to launch our first vent as an assembled unit. I've listed the 4x10 for now, but will work on adding the rest shortly. If you are looking for a specific size (option from our list in the readme), please let me know and I'll prioritize those.

Our price is little bit higher than I original anticipated; I hope that as we continue down this path we'll find some savings on cost and pass that on to the customer.

Let me know if you have any questions or concerns.

https://brobston-creations.mybigcommerce.com/yet-another-smart-vent-4x10/

**Comment by TonyBrobston** [comment]
> > @Daniel-dev22 @veaceslavc @AwokenByMyFears
> > My apologies for the radio-silence. I had taken a break from this project for a few months. In the last few weeks I've started working on it again.
> > We're excited to launch our first vent as an assembled unit. I've listed the 4x10 for now, but will work on adding the rest as I have time. If you are looking for a specific size (option from our list in the readme), please let me know and I'll prioritize those.
> > Our price is little bit higher than I original anticipated; I hope that as we continue down this path we'll find some savings on cost and pass that on to the customer.
> > Let me know if you have any questions or concerns.
> > https://brobston-creations.mybigcommerce.com/yet-another-smart-vent-4x10/
> 
> Will these come pre-flashed with esphome out of the box?

Yes.

It should be as simple as: Power the vent on, connect the access point it broadcasts, enter your wifi credentials, assuming it successfully connects, and mDNS works correctly on your network, the vent should be discovered by Home Assistant.

**Comment by TonyBrobston** [comment]
> What's your current lead time to ship?

Right now these are made to order, so it depends on how many are ordered.

To error on the safe side, I'd say from the time of ordering to dropped off to USPS it would be about a vent a day. Probably slightly longer for smaller orders and slightly short for larger orders. As we start to market these and see more orders, we will start to manufacture the more common sizes prior to the order.

> Also does it have a built in temperature sensor?

It does not. This is something I contemplated and at least for now, decided against.

My thinking is that the temperature at the vent is not really a good representation of the temperature of the room. Right now I have one Shelly H&T (https://www.shelly.com/en-us/products/shop/shelly-h-t-white/shelly-h-t-black) in each room; while these are somewhat affordable there are certainly less expensive options.

**Comment by TonyBrobston** [comment]
> Yeah it's not a good representation of the temperature of the room but it's useful as a measure or the temperature/moisture coming out of the register.

I think that's fair.

I'll experiment with adding a temperature sensor. I'd like to use the same temperature sensor I'm using in this project's sibling project (https://github.com/BrobstonCreations/delta-tango), the DS18B20.

If I went the direction of temperature and humidity, I would have to research to see if I can find a sensor that's viable.

I picture this temperature sensor coming out of the bottom of the vent, to measure duct/register air temperature. Though I could also picture it coming out the top, though that feels a bit wonky as it would be unclear whether it is meant to read duct air temperature or the room air temperature, when the intent I believe is to measure the duct air temperature.

Let me know your thoughts on this.

**Comment by jlk** [comment]
I've had good luck with DHT22s over the years. Looks like esphome supports it out of the box https://esphome.io/components/sensor/dht.html

**Comment by TonyBrobston** [comment]
I went ahead and order some of these to experiment with: https://esphome.io/components/sensor/aht10

I think the biggest difficulty is figuring out where to put it, but I'll play around in CAD and see what I can come up with. I'll let you know as I have some results. Probably in a few days, but could be longer.

**Comment by TonyBrobston** [comment]
> Awesome interested in the results. Hopefully it works out and I'll place an order in the next 2 weeks to get 2 vents to play after I hopefully sell these keen vents.

I was able to find a few minutes to hook this sensor up to a D1 Mini and was able to use Home Assistant to read temp and humidity. The next step is to figure out how to fit in in/on the vent.

**Comment by TonyBrobston** [comment]
> Awesome. Hopefully it works out to integrate it into the setup, would this sensor be offered as an additional option on the site?

I think this will just become a standard part of the vent. Seems valuable the more I think about it; it's a little bit more cost and effort to install, but otherwise I think it's a positive addition.

**Comment by TonyBrobston** [comment]
Small update. I finished the revisions to add the temp/humidity sensor to part of the electronics enclosure.

Not everything is modeled, but you can see where the where the temp/humidity sensor would be exposed to the air coming from the vent duct.
![image](https://github.com/BrobstonCreations/yet-another-smart-vent/assets/4724577/a5b4485c-c02a-4d85-9224-76e4da20ddb9)
![image](https://github.com/BrobstonCreations/yet-another-smart-vent/assets/4724577/5d66890c-f3b5-4733-9976-5ef9edfd1551)

I'll hopefully get one of these printed later today. I've only done a preliminary test of the temp/humidity sensor and in theory it should integrate fine with the other electronics; but I'll need to test before I can 100% confirm. I'll hopefully have this done in the next few days, I'll update you when I do.

**Comment by TonyBrobston** [comment]
@Daniel-dev22 Sorry for the slow reply. I've had a lot going on between work and other projects.

I took some time this morning to work on this again. I haven't put the electronics into the vent, though I did mock them up.

More importantly, I got the two ESPHome configurations merged together and flashed to this chip and everything seems to be working as expected.

![image](https://github.com/BrobstonCreations/yet-another-smart-vent/assets/4724577/12163dd9-b734-440e-8bb2-907cdaed9588)

**Comment by TonyBrobston** [comment]
> @TonyBrobston any updates on the vents? I am ready to buy 2 white 4x10s to start with, that have the temperature sensors integrated.

I'm currently on a trip. The earliest I'd get to things is the last week of December. I don't know of exactly when I will, but I do plan to. I imagine early January. Unfortunately business stuff hasn't been a huge priority lately, but I am interested in producing some and see where it goes.

**Comment by hidden0** [comment]
Just wanted to say it's awesome this email thread is still going.
I asked you some questions forever ago and still occasionally check in on
things.

**Comment by skl111** [comment]
@TonyBrobston I am finally at the point where I want to replace all my supply vents with smart vents and I don't want to use flair.
I have a few questions.
Have you thought about adding a version that uses fans to boost the air flow in certain rooms?
Have you thought about adding vertical louvers?
Do you plan to sell pre-assembled hardware kits excluding the vent & louvers?

**Comment by ssalahi** [comment]
Hi @TonyBrobston ,
Your store seems to be down. Any plan to bring it back online?
I an interested in 10x4 went and don’t care about the temp sensor. (With/without it will work)
Or even, the parts kit would be great as I have 3D printer myself and can do the rest. (Although prefer not to 😄)

I really prefer not to buy parts one by one from different locations!

**Comment by TonyBrobston** [comment]
> @TonyBrobston
> 
> If I buy 2 4x10 vents off the site even without the extra temperature/humidity sensors added can they be shipped?

He Daniel, sorry for the delay. If you go ahead and order these I'll get them put together and sent to you.

**Comment by TonyBrobston** [comment]
> Hi @TonyBrobston , Your store seems to be down. Any plan to bring it back online? I an interested in 10x4 went and don’t care about the temp sensor. (With/without it will work) Or even, the parts kit would be great as I have 3D printer myself and can do the rest. (Although prefer not to 😄)
> 
> I really prefer not to buy parts one by one from different locations!
@ssalahi 
Sorry about that, the store is back up.

It is up to you how you'd like to order things. The options that are on the site are currently what I'm offering. I plan to expand the sizes at some point; I have all the work done, I just haven't posted the option to the store. I'm not currently offering to assemble the electronics, so that kit comes disassembled, but if you order a whole vent that will come assembled.

Let me know if you have any other questions.

**Comment by ssalahi** [comment]
> > Hi @TonyBrobston , Your store seems to be down. Any plan to bring it back online? I an interested in 10x4 went and don’t care about the temp sensor. (With/without it will work) Or even, the parts kit would be great as I have 3D printer myself and can do the rest. (Although prefer not to 😄)
> > I really prefer not to buy parts one by one from different locations!
> 
> @ssalahi Sorry about that, the store is back up.
> 
> It is up to you how you'd like to order things. The options that are on the site are currently what I'm offering. I plan to expand the sizes at some point; I have all the work done, I just haven't posted the option to the store. I'm not currently offering to assemble the electronics, so that kit comes disassembled, but if you order a whole vent that will come assembled.
> 
> Let me know if you have any other questions.

Hi @TonyBrobston, 
I am not sure how to contact you outside of here as I replied to the email from the store once and I don't think it was ever reviewed. 
I have ordered one to check and order multiple later and these are my comments so far: 
Pros: 
1- The device was easy to setup
2- ESPHome did detect the device without issue and I am able to open/close without any problem
3- The device has Temp/Humidity sensor (which I didn't know, I guess I would have if I had read the whole thing properly)
4- It looks pretty good and it's not loud when you open/close it
5- Having the option of 2 different power source type is amazing.

Cons: 
1- It doesn't close all the way and stays at 98% max when you use motor to close it and when air is going through, it causes to makes an annoying sound
2- My house doesn't have the ceiling cut pretty well which causes me to actually not being able to mount it as it screw locations are pretty close to the side.
3- Not sure what temperature is being read, but it's definitely off 😄 

Is it possible for you to create an outer layer model that is similar to typical vent in terms of size and holes? this will help me to be able to secure it properly. I have a 3D printer and can print that part myself :) 

Something similar to this product, not in term of side air flow, but in term of size and hole placement (although that would have been nice to have as well!): 
https://www.amazon.com/dp/B0B256W8F8/
![Sample Image](https://m.media-amazon.com/images/I/81HDQV+4GWL._AC_SL1500_.jpg)

**Comment by TonyBrobston** [comment]
> > > Hi @TonyBrobston , Your store seems to be down. Any plan to bring it back online? I an interested in 10x4 went and don’t care about the temp sensor. (With/without it will work) Or even, the parts kit would be great as I have 3D printer myself and can do the rest. (Although prefer not to 😄)
> > > I really prefer not to buy parts one by one from different locations!
> > 
> > 
> > @ssalahi Sorry about that, the store is back up.
> > It is up to you how you'd like to order things. The options that are on the site are currently what I'm offering. I plan to expand the sizes at some point; I have all the work done, I just haven't posted the option to the store. I'm not currently offering to assemble the electronics, so that kit comes disassembled, but if you order a whole vent that will come assembled.
> > Let me know if you have any other questions.
> 
> Hi @TonyBrobston, I am not sure how to contact you outside of here as I replied to the email from the store once and I don't think it was ever reviewed. I have ordered one to check and order multiple later and these are my comments so far: Pros: 1- The device was easy to setup 2- ESPHome did detect the device without issue and I am able to open/close without any problem 3- The device has Temp/Humidity sensor (which I didn't know, I guess I would have if I had read the whole thing properly) 4- It looks pretty good and it's not loud when you open/close it 5- Having the option of 2 different power source type is amazing.
> 
> Cons: 1- It doesn't close all the way and stays at 98% max when you use motor to close it and when air is going through, it causes to makes an annoying sound. Is it possible to force it to fully close? Do I need to adjust something manually? (I feel it needs some code update to force the motor to have higher power) 2- My house doesn't have the ceiling cut pretty well which causes me to actually not being able to mount it as it screw locations are pretty close to the side. 3- Not sure what temperature is being read, but it's definitely off 😄
> 
> Is it possible for you to create an outer layer model that is similar to typical vent in terms of size and holes? this will help me to be able to secure it properly. I have a 3D printer and can print that part myself :)
> 
> Something similar to this product, not in term of side air flow, but in term of size and hole placement (although that would have been nice to have as well!): https://www.amazon.com/dp/B0B256W8F8/ ![Sample Image](https://camo.githubusercontent.com/3be4f16983778ae041da9fb22b87afaf97b148a33e3fd0fd69e09427c8092afb/68747470733a2f2f6d2e6d656469612d616d617a6f6e2e636f6d2f696d616765732f492f3831484451562b3447574c2e5f41435f534c313530305f2e6a7067)
> 
> BTW, thanks for creating this great product!!

Hey Saeed,

Sorry for the delay. I need to look into where those emails go when replying to Big Commerce, they didn't come through to me. I emailed you directly. We very much appreciate the feedback and will work on design improvements and likely send you out a revised version to fix the issues.

Thanks,
Tony

**Comment by jmunchies1** [comment]
Hi @TonyBrobston

Are premade vents no longer being sold? The link from this thread is dead and I can't find any on the website. 

I am just looking to purchase one vent.

Thanks!
